tests/test_diffraction_objects.py

```diff
@@ -165,99 +167,108 @@ def test_init_invalid_xtype():
+                "yarray": np.array([2, 3, 4, 5, 6, 7]),
+                "xtype": "tth",
+                "wavelength": 2 * np.pi,
+            },
+            {
+                "q": None,
+                "tth": 60,
+                "d": None,
+                "offset": 2.1,
+            },
+            {"xtype": "tth", "yarray": np.array([4.1, 5.1, 6.1, 7.1, 8.1, 9.1])},
+        ),
+        # UC2: same length x-arrays with exact x-value match
+        (
+            {
+                "xarray": np.array([10, 15, 25, 30, 60, 140]),
+                "yarray": np.array([10, 20, 25, 30, 60, 100]),
+                "xtype": "tth",
+                "wavelength": 2 * np.pi,
+            },
+            {
+                "xarray": np.array([10, 20, 25, 30, 60, 140]),
+                "yarray": np.array([2, 3, 4, 5, 6, 7]),
+                "xtype": "tth",
+                "wavelength": 2 * np.pi,
+            },
+            {
+                "q": None,
+                "tth": 60,
+                "d": None,
+                "offset": 0,
+            },
+            {"xtype": "tth", "yarray": np.array([1, 2, 2.5, 3, 6, 10])},
+        ),
+        # UC3: same length x-arrays with approximate x-value match
+        (
+            {
+                "xarray": np.array([0.12, 0.24, 0.31, 0.4]),
+                "yarray": np.array([10, 20, 40, 60]),
+                "xtype": "q",
+                "wavelength": 2 * np.pi,
+            },
+            {
+                "xarray": np.array([0.14, 0.24, 0.31, 0.4]),
+                "yarray": np.array([1, 3, 4, 5]),
+                "xtype": "q",
+                "wavelength": 2 * np.pi,
+            },
+            {
+                "q": 0.1,
+                "tth": None,
+                "d": None,
+                "offset": 0,
+            },
+            {"xtype": "q", "yarray": np.array([1, 2, 4, 6])},
+        ),
+        # UC4: different x-array lengths with approximate x-value match
+        (
+            {
+                "xarray": np.array([10, 25, 30.1, 40.2, 61, 120, 140]),
+                "yarray": np.array([10, 20, 30, 40, 50, 60, 100]),
+                "xtype": "tth",
+                "wavelength": 2 * np.pi,
+            },
+            {
+                "xarray": np.array([20, 25.5, 32, 45, 50, 62, 100, 125, 140]),
+                "yarray": np.array([1.1, 2, 3, 3.5, 4, 5, 10, 12, 13]),
+                "xtype": "tth",
+                "wavelength": 2 * np.pi,
+            },
+            {
+                "q": None,
+                "tth": 60,
+                "d": None,
+                "offset": 0,
+            },
+            # Scaling factor is calculated at index = 4 (tth=61) for self and index = 5 for target (tth=62)
+            {"xtype": "tth", "yarray": np.array([1, 2, 3, 4, 5, 6, 10])},
+        ),
+    ],
+)
+def test_scale_to(org_do_args, target_do_args, scale_inputs, expected):
+    original_do = DiffractionObject(**org_do_args)
```

Comment:
then uses `**org_do_args` to init `DO`. which looks like a better solution than 

```
    orig_diff_object = DiffractionObject(
        xarray=inputs["xarray"], yarray=inputs["yarray"], xtype=inputs["xtype"], wavelength=inputs["wavelength"]
    )
    target_diff_object = DiffractionObject(
        xarray=inputs["target_xarray"],
        yarray=inputs["target_yarray"],
        xtype=inputs["target_xtype"],
        wavelength=inputs["target_wavelength"],
    )
```

Reply:
yes, better and better :+1: